test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresUnreferencedCodeCapability.cs

```diff
@@ -778,8 +778,7 @@ class NewConstrainTestType
 
 			static void GenericMethod<T> () where T : new() { }
 
-			// https://github.com/mono/linker/issues/2117
-			[ExpectedWarning ("IL2026", "--NewConstrainTestType.ctor--", GlobalAnalysisOnly = true)]
+			[ExpectedWarning ("IL2026", "--NewConstrainTestType.ctor--")]
```

Comment:
We should also add a test when the new constraint is on a type parameter of a type (this one is on method).
For linker there's basically no difference, but for the analyzer there is a big one. I don't know if the analyzer will already handle this or not. Feel free to add it in a separate PR.

Reply:
Added a test and made the changes in this same PR. PTAL.